### PR TITLE
Create SpaServices Module

### PR DESCRIPTION
**Pull request by MichaelPetrinolis**
This PR allows to serve a Single Page Application per tenant.
The user can bootstrap the SPA from a static file (eg.index.html) or a liquid template.
Added the SampleDataController in OrchardCore.Demo that is used by the dotnet spa templates
in order to work OOB (for someone testing the orchardcore). I also added a reference to it in OrchardCore.CMS.Web, Also added a recipe that enables the SpaServices using the static file option

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.SpaServices/OrchardCore.SpaServices.csproj [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" />
+    <PackageReference Include="Microsoft.AspNetCore.SpaServices.Extensions" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentManagement.Abstractions\OrchardCore.ContentManagement.Abstractions.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentManagement.Display\OrchardCore.ContentManagement.Display.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Module.Targets\OrchardCore.Module.Targets.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Navigation.Core\OrchardCore.Navigation.Core.csproj" />
+    <ProjectReference Include="..\OrchardCore.Recipes\OrchardCore.Recipes.csproj" />
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.SpaServices/OrchardCore.SpaServices.csproj]: We don't reference modules from other modules

Comment (MichaelPetrinolis) [reply src/OrchardCore.Modules/OrchardCore.SpaServices/OrchardCore.SpaServices.csproj]: fixed the references.

**Comment by agriffard** [comment]
The name SpaServices could be confusing from my point of view.
It has to be clear for the user when he chooses the recipe, whose name is `Headless` btw.
These are a lot of various concepts and I can't really see the consistency between them.

It seems to be a one page site based on a static file and editable with a template.

**Comment by MichaelPetrinolis** [comment]
> The name SpaServices could be confusing from my point of view.
> It has to be clear for the user when he chooses the recipe, whose name is `Headless` btw.

I agree, SpaServices is for developing/hosting a spa, this modules sets the homepage to the SPA. 
What about OrchardCore.SPAHomepage or OrchardCore.Headless ?

> These are a lot of various concepts and I can't really see the consistency between them.
> 
> It seems to be a one page site based on a static file and editable with a template.

Indeed, i thought that would be helpful to have an easy way to create a tenant and host a Single Page Application for the headless scenario. It misses a way to update the SPA from admin, but it would be easy to add this functionality.

I have tested it with oob dotnet new  for angular, aurelia, react and vue and one must only set the <base href="/tenantprefix/"> in order to work. The routing works fine and it fallback to SPA on a not found result

**Comment by Skrypt** [comment]
I was thinking. Is it obligated that we have a checkbox to enable/disable the index.html file option? Could we not just display an error message if there is no index.html or template found and if a template is found it would just take precedence on the index.html file or vice versa?

**Pull request closed without merging**